test/utils/runners.go

```diff
@@ -1493,7 +1508,24 @@ func NewCustomCreatePodStrategy(podTemplate *v1.Pod) TestPodCreateStrategy {
 // volumeFactory creates an unique PersistentVolume for given integer.
 type volumeFactory func(uniqueID int) *v1.PersistentVolume
 
-func NewCreatePodWithPersistentVolumeStrategy(claimTemplate *v1.PersistentVolumeClaim, factory volumeFactory, podTemplate *v1.Pod) TestPodCreateStrategy {
+// PodTemplate is responsible for creating a v1.Pod.
```

Comment:
Note that this is conceptually similar to `volumeFactory` above, I just decided to use a template instead of a function. That interface then has two different implementations:
- transforming a YAML file with text/template (new)
- returning a fixed pod instance (what was done before)

If you like the interface approach, then a follow-up PR could use the same concept also for volumes. Not sure whether it's worth it, though.

Reply:
We can touch that part once needed.